### ISSUE TITLE
Links in combined PDFs are still broken

### ISSUE THREAD

**Issue by erilot** [zhaoterryy/mkdocs-pdf-export-plugin#43]
Running mkdocs-pdf-export-plugin 0.5.3, relative links between pages are still pointing to local file locations:

![image](https://user-images.githubusercontent.com/2288230/53054205-7bc84680-3458-11e9-87a5-4353c4cb4762.png)

It looks like this was fixed in #28 but may have regressed since then. I've tested this in my existing projects and in a clean test project (as shown in the screenshot above, relative links are all pointing to `file:///C:...` locations. The links work properly in rendered HTML.

During the build, the same error gets thrown four times (the test project has only 3 pages), which seems like the same thing experienced in #42:

```
ERROR: No anchor #/.:pdf-export-test for internal URI reference
ERROR: No anchor #/.:pdf-export-test for internal URI reference
ERROR: No anchor #/.:pdf-export-test for internal URI reference
ERROR: No anchor #/.:pdf-export-test for internal URI reference
```

There is no file called pdf-export-test in the test project, but that is the H1 on the index.md page.

I'd be happy to provide the test project if it would be helpful. I could take a stab at a PR but I'm still learning my way around the code (and know nothing about WeasyPrint, unfortunately).

And lastly, this project is a lifesaver! Thank you so much for your work and active support. It is greatly appreciated.

**Comment by CaptainQuirk** [comment]
On my part, trying out the `0.5.3` version didn't solve my problem in #42.

I still get `No anchor #cp/show/:charge-de-communication for internal URI reference` for the markdown construct `En tant que [Chargé de communication](../glossaire.md#charge-de-communication)`

:+1: for the appreciation of the whole project

**Comment by zhaoterryy** [comment]
> I'd be happy to provide the test project if it would be helpful. I could take a stab at a PR but I'm still learning my way around the code (and know nothing about WeasyPrint, unfortunately).

Yes please provide an example, my project here doesn't reproduce this issue. Thank you!

**Comment by erilot** [comment]
Here's the test project I'm using. I was curious if it was a Windows issue so opened the project on a Linux/Ubuntu desktop but am seeing the same behavior; PDFs are created successfully but links are pointing to local files instead of locations within the PDF. I also get errors for each head tag in the project, as above (this one is slightly different than my original report because I re-created everything from scratch).

```
ERROR: No anchor #/.:pdf-export-test-project for internal URI reference
ERROR: No anchor #/.:pdf-export-test-project for internal URI reference
ERROR: No anchor #/.:pdf-export-test-project for internal URI reference
ERROR: No anchor #/.:pdf-export-test-project for internal URI reference
ERROR: No anchor #/.:subheading for internal URI reference
```
Other details that may be useful:

- Weasyprint: 45
- mkdocs-pdf-export-plugin: 0.5.3
- mkdocs: 1.0.4
- mkdocs-material: 4.0.1 (also tested with 3.2.0)
- Linux/Ubuntu 18.04 (also tested with Windows 10)

Project:
[test.zip](https://github.com/zhaoterryy/mkdocs-pdf-export-plugin/files/2895670/test.zip)

Generated PDF:
[combined.pdf](https://github.com/zhaoterryy/mkdocs-pdf-export-plugin/files/2895674/combined.pdf)

Verbose output (or most of it anyway, it filled up my buffer so we're missing the beginning)
[console-output.txt](https://github.com/zhaoterryy/mkdocs-pdf-export-plugin/files/2895678/console-output.txt)

Thanks again for looking into it!

**Comment by zhaoterryy** [comment]
I found the issue. For the time being, `use_directory_urls` must be set to false. https://www.mkdocs.org/user-guide/configuration/#use_directory_urls

Need to write a separate href builder for directory urls.

**Comment by erilot** [comment]
Sorry to drop this, was away for a couple of weeks!

This workaround fixes one problem -- I no longer get pointed at local files -- but it looks like there was another issue hiding behind that one.

Now that the local file links are fixed (or worked around), some links work, but some links just render as styled text that doesn't go anywhere.

* Links in pages at the root `docs/` level, **to other pages at the same level**, work as expected 🎉.
* Links in pages at the root `docs/` level with anchors, like `[link](file.md#heading)`, or with *only* anchors, like `[link](#heading)`, do not work.
* Links in pages under subfolders (for example, `docs/folder/file.md`) seem to never work no matter what.

I've expanded my test project to show these individual cases.  If I can add test cases that would be helpful for you, I'm happy to build it out further. Also if this looks like a totally different issue to you, we can close this off and start a new one.

Revised test project: [pdf-link-test-project.zip](https://github.com/zhaoterryy/mkdocs-pdf-export-plugin/files/2954634/pdf-link-test-project.zip)

Rendered PDF: [test.pdf](https://github.com/zhaoterryy/mkdocs-pdf-export-plugin/files/2954639/test.pdf)

Build output (not verbose this time because whoa): [output.txt](https://github.com/zhaoterryy/mkdocs-pdf-export-plugin/files/2954644/output.txt)

**Comment by zhaoterryy** [comment]
@erilot try updating to v0.5.5, that should fix those issues.

**Comment by majkinetor** [comment]
Not only broken, but unclickable.

**Comment by CaptainQuirk** [comment]
@zhaoterryy @erilot any news on this ?

I'm still having issues with links containing anchors and several `../` in their paths

**Comment by erilot** [comment]
@CaptainQuirk sorry, forgot to watch my notifications. I cobbled together a solution for this (actually completely rebuilt the path generator) and it works for our team, but I wasn't able to submit a PR here and then forgot about it.

I've forked this project and updated it with my code, you're welcome to use it (the fork is not registered with pypi so you can't use pip to install it, you'll need to install it manually or copy the plugin files over this install.

https://github.com/erilot/mkdocs-pdf-export-plugin

@zhaoterryy I would be happy to submit this as a PR here but i think you have to allow that.

**Comment by zhaoterryy** [comment]
@erilot You should be able to submit a PR, I've accepted other's before. If you're still unable to then I can manually create one myself from your repo.

Thanks!!

**Comment by erilot** [comment]
@zhaoterryy you're totally right, I don't know what I did wrong before.

PR is here: https://github.com/zhaoterryy/mkdocs-pdf-export-plugin/pull/64

**Comment by majkinetor** [comment]
FYI, I created docker mkdocs framework which uses this plugin among other things.

* https://github.com/majkinetor/mm-docs-template

You can view generated demo output here:

* https://majkinetor.github.io/mm-docs-template/demo

and [download pdf](https://majkinetor.github.io/mm-docs-template/demo/demo.pdf)

You can see that first links aren't clickable but later on second page so called magic links are clickable.

Latest plugin version.

**Comment by igorcavalcante** [comment]
> I found the issue. For the time being, `use_directory_urls` must be set to false. https://www.mkdocs.org/user-guide/configuration/#use_directory_urls
> 
> Need to write a separate href builder for directory urls.

Thanks, it solved my problem today!

**Comment by fletort** [comment]
ANy news ? I used the use_directory_urls to false, but the error is always here for subdirectory link....

**Comment by FlorianHeigl** [comment]
The PR seems stuck... I also still get this kind of problems (anchor used in file in subfolder), too tired to try out why the patch has a conflict :(
